hw3/02-charts.js

```diff
@@ -29,24 +31,123 @@ const borderColors = [
 // url for the Thrones API
 const url = 'https://thronesapi.com/api/v2/Characters';
 
-const renderChart = () => {
+
+// async function to fetch the data from a given url
+const fetchData = async function fetchDataFromApi(url) {
+  try {
+      const res = await fetch(url);
+      const data = await res.json();
+      return data;
+  }
+  catch {
+      console.error('Error', error);
+  }
+};
+
+// function to count family names
+const getFam = function getFamilyNames(data) {
+
+  let familyNames = [];
+
+  data.forEach(element => {
+    if(!familyNames.some(name => name.name === element.lastName)) {
+      familyNames.push(
+        {
+          'name': element.lastName,
+          'count': 1
+        });
```

Comment:
This seems to be emulating a map with an array. Since every `findIndex` or `some` will have to iterate over many elements of the array, it can be quite inefficient.

I recommend using the `Map` class from the standard library instead: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

Reply:
Considered it, but I haven't used them enough to implement within my schedule. maybe next time!